satpy/readers/netcdf_utils.py

```diff
@@ -196,6 +202,11 @@ def __getitem__(self, key):
                 val = self._get_var_from_filehandle(group, key)
             else:
                 val = self._get_var_from_xr(group, key)
+        elif isinstance(val, netCDF4.Group):
+            # Full groups are conveniently read with xr even if file_handle is available
+            with xr.open_dataset(self.filename, group=key,
+                                 **self._xarray_kwargs) as nc:
+                val = nc
```

Comment:
Will the context manager not close the nc file? Will we not loose access to the data if we use dask in the background?

Reply:
From my tests, the problem does not happen. Data can be correctly read.